packages/adapters/src/adapters/Cornerstone3D/Segmentation/generateLabelMaps2DFrom3D.ts

```diff
@@ -0,0 +1,52 @@
+function generateLabelMaps2DFrom3D(labelmap3D) {
+    // 1. we need to generate labelmaps2D from labelmaps3D, a labelmap2D is for each
+    // slice
+    const { scalarData, dimensions } = labelmap3D;
+
+    // scalarData is a flat array of all the pixels in the volume.
+
+    const labelmaps2D = [];
+    const segmentsOnLabelmap3D = new Set();
+
+    for (let z = 0; z < dimensions[2]; z++) {
+        const pixelData = scalarData.slice(
+            z * dimensions[0] * dimensions[1],
+            (z + 1) * dimensions[0] * dimensions[1]
+        );
+
+        const segmentsOnLabelmap = [];
+
+        for (let i = 0; i < pixelData.length; i++) {
+            const segment = pixelData[i];
+            if (!segmentsOnLabelmap.includes(segment) && segment !== 0) {
+                segmentsOnLabelmap.push(segment);
+            }
+        }
+
+        const labelmap2D = {
+            segmentsOnLabelmap,
+            pixelData,
+            rows: dimensions[1],
+            columns: dimensions[0]
+        };
+
+        if (segmentsOnLabelmap.length === 0) {
+            continue;
+        }
+
+        segmentsOnLabelmap.forEach(segmentIndex => {
+            segmentsOnLabelmap3D.add(segmentIndex);
+        });
+
+        labelmaps2D[dimensions[2] - z] = labelmap2D;
```

Comment:
Should that be `dimensions[2] - 1 - z`?

Reply:
Is this generating label maps that are frame indexed starting with 1?  If so, that fact should be in a comment here and should be noted up top.